python/pyspark/pandas/tests/indexes/test_base.py

```diff
@@ -203,9 +203,35 @@ def test_to_frame(self):
         # non-string names
         self.assert_eq(psidx.to_frame(name=[10, 20]), pidx.to_frame(name=[10, 20]))
         self.assert_eq(psidx.to_frame(name=("x", 10)), pidx.to_frame(name=("x", 10)))
-        self.assert_eq(
-            psidx.to_frame(name=[("x", 10), ("y", 20)]), pidx.to_frame(name=[("x", 10), ("y", 20)])
-        )
+        if LooseVersion(pd.__version__) < LooseVersion("1.5.0"):
+            self.assert_eq(
+                psidx.to_frame(name=[("x", 10), ("y", 20)]),
+                pidx.to_frame(name=[("x", 10), ("y", 20)]),
+            )
+        else:
+            # Since pandas 1.5.0, the result is changed as below:
+            #      (x, 10)  (y, 20)
```

Comment:
What's the exact diff between <1.5 and >=1.5?

Reply:
mixed type in its name? or value? what's the mixed type here?

Reply:
The exact diff is shown as below:

```python
# pandas < 1.5.0
>>> pd.Index([('x', 10), ('y', 20)]).to_frame(name=[("x", 20), ("y", 20)])
      x   y
     20  20
x 10  x  10
y 20  y  20

# pandas >= 1.5.0
>>> pd.Index([('x', 10), ('y', 20)]).to_frame(name=[("x", 20), ("y", 20)])
     (x, 20)  (y, 20)
x 10       x       10
y 20       y       20
```

The column should be like `Index([('x', 20), ('y', 20)], dtype='object')`, but such as way is not allowed in pandas API on Spark:

```python
>>> ps.Index([['x', 20], ['y', 20]])
Traceback (most recent call last):
...
pyarrow.lib.ArrowTypeError: Expected bytes, got a 'int' object
```

Reply:
Let me fix the comment to more explicitly.